### DISCUSSION TITLE
Qubino ZMNHID1 Thermostat not working properly (FW 1.1)

### DISCUSSION THREAD

**Discussion by kurdybanek** [zwave-js/zwave-js#7300] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

I have few ZMNHID1 Qubino Thermostats (Home Assistant with Z-wave JS UI). Most of them have firmware in a version 5.x - they work properly.
I also have one unit with old firmware - 1.1, which is recognized and configured during the pairing, but I can't set the temperature using this device. 
I can switch it on/off, I see temperature readings from attached temp sensor, I see power usage - only temperature setpoint is not working. In UI I see the dot on the circle presenting temperate range, but I can't move it. I can't change it also directly from the Z-wave JS UI, via the "Thermostat setpoint v2" section.
It also shows in a different way than other units in the UI - it has OFF and Heat/Cool button, while other ones have OFF and Heat buttons.
I know the device itself is working properly, as I paired with an Vera unit I have and didn't have any issues with setting temperature and other functionalities.

I would like to be able to change temperature setpoint.

To reproduce behavior:
Go to UI, turn thermostat on, try to change temperature setpoint.

Comparing this unit with other ones I have it seems that the definition of the setpoint is wrong. 
For other units, in the "Themperature setpoint v2" section in Z-wave JS UI I see heading: "**[6-67-0-setpoint-1] Setpoint (Heating)**" (first number is Node ID). In the one which works incorrect I see: **[12-67-0-setpoint-0] Setpoint (N/A)**. The last value is "0" instead of "1" and I assume that it is not recognized by the unit.
Under the Thermostat Mode v2 section I see that parameter **[12-64-0-mode] Thermostat mode** can be set to [0]OFF or [3]Auto, while for other units it is [0]OFF and [1]Heat, though setting the mode works correctly (the unit turns on or off depending of the settings)

### Software versions

Driver (node-zwave-js): 13.9.1

Z-Wave JS UI: 3.16.0

Home Assistant Z-Wave Integration: Z-wave JS

Home Assistant Z-Wave JS Addon: .Z-wave js UI

ioBroker.zwave2 Adapter:  

If you are using something non-standard, tell us here: ...

### Z-Wave Controller (Stick/Dongle/...)

Currently set on testing instance with Aeotec Z-Stick 7 (Firmware 7.21.4, SDK 7.21,4), but also tested with Zooz ZST39 and Aeotec Z-Stic Gen5+ 

### Device information

Manufacturer: Qubino
Model name: Flush Thermostat, ZMNHID1
Node ID: 12

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/user-attachments/files/17466123/zwave-js-ui-store.zip)
[node_12_dump.json](https://github.com/user-attachments/files/17466126/node_12_dump.json)

**Comment by kurdybanek** [comment]
I would try to use custom device configuration file but I can't find how I can download the thermostat configuration file which is currently used by this device (I would prefer to use existing template as a base rather than creating something completely new).
I downloaded configuratio file for ZMNHID from the Z-Wave JS Config DB Browser, but it is mostly about configuration parameters, not the definition of the device type and realated options (like heating., setpoint, etc.)

**Comment by AlCalzone** [comment]
Please re-interview the device and share a log of that.

> but it is mostly about configuration parameters, not the definition of the device type and realated options (like heating., setpoint, etc.)

That is because Z-Wave JS relies on the information reported by the device to learn what it is and what it can do.

**Comment by kurdybanek** [comment]
OK, I did re-interview and downloaded the logs folder (attached).
The unit is currently paired with my testing instance, so I can perform other activities if needed

[zwave-js-ui-store (1).zip](https://github.com/user-attachments/files/17474583/zwave-js-ui-store.1.zip)

**Comment by kurdybanek** [comment]
By the way, I found some issue with temp probe (so in log you may find temp reading switching to 85 degrees and back to the real temp.). I replaced the sensor with working one, so it now doesn't jump so if you want me to re-interview device once more, I can do it.

**Comment by AlCalzone** [comment]
Looks like that device incorrectly reports which setpoints it supports. Do you have a manual or something that states which setpoints should actually be supported?

**Reply by kurdybanek** [reply]
I don't, but I can try to reach to Qubino for that (if they are not closed after takeover by Shelly).
I found some info on the OpenZWAVE github but I'm not sure if it will be helpful.

"..._Command Class 64:
this model (H1S2P1) has two modes: off and Auto, and its value are 0 and 3 (not 1). Yes it's a strange value, but it works with this values, not with 1 (it looks like it uses the bit 1 and not bit 0)_"
 
link to discussion: https://github.com/OpenZWave/open-zwave/issues/1857

If I understand above correctly, the proper z-wave command should be xx-67-0-setpoint-**3** instead of xx-67-0-setpoint-0 (where "xx" is the node number)?
But when I try to set up an automation using such command, it returns an error (see attached screenshots). When I change property key to 0 it does not return an error, but nothing happens

I can also pair this device again with my old Vera (where it works without issues) and extract any configuration info from it - as far as I remember there were some json configuration files for devices. Shall I do it?

![image](https://github.com/user-attachments/assets/dcbaa623-9832-465d-bc0a-aec7544b2597)
![image](https://github.com/user-attachments/assets/20aeef74-5d0a-4c9d-84c9-43b99a7e5e1c)

**Reply by AlCalzone** [reply]
Thermostat modes and setpoint types are not necessarily the same. I actually expect the device to have a "Heating" setpoint, but we need to verify before making any changes.

**Reply by kurdybanek** [reply]
OK, I've got a manual file for this specific version from Qubino (see attached).
There are difference compared to newer revisions - for example there is no parameter 59 (Thermostat mode) - that explains why it was empty when I entered configuration section. 
Since they are replying promptly, should I ask them for something more specific?
[Qubino_Flush On Off Thermostat PLUS_user manual_V1.1_eng.pdf](https://github.com/user-attachments/files/17477515/Qubino_Flush.On.Off.Thermostat.PLUS_user.manual_V1.1_eng.pdf)

**Reply by kurdybanek** [reply]
I found some other interesting thing. If I look into z-wave.js UI device panel and into the configuration v2 section, I see that the configuration parameters not available for this device (but supposedly taken from the configuration file from never versions) are at the sorted at the end (regardless of their number) with blank values (only parameter 59 has a value, but this is after I tried to change it):

![image](https://github.com/user-attachments/assets/dc22316b-d5c1-4c14-ad94-78506938e4fe)

From the other hand there is missing parameter 30, which is mentioned in the manual I've attached in the post above

**Comment by kurdybanek** [comment]
Hi, any chance to solve my issue?

**Comment by kurdybanek** [comment]
Bump?
